cluster-creation-cli/README.md

```diff
@@ -0,0 +1,334 @@
+---
+creation_date: 2024-05-28
+issues:
+- https://github.com/giantswarm/roadmap/issues/2115
+owners:
+- https://github.com/orgs/giantswarm/teams/team-honeybadger
+state: review
+summary: Specification of a kubectl-gs command for creating a workload cluster in the Cluster API generation of the Giant Swarm platform
+---
+
+# CLI command for creating a new cluster
+
+This RFC proposes the spec of a new command line interface (CLI) command to create a workload cluster in the Giant Swarm platform based on the Kubernetes Cluster API.
+
+## Terminology
+
+- **CLI** or **command line interface** refers to a text-based user interface to interact with a computer program.
+
+- **Configuration** in the context of this RFC refers to the configuration of a workload cluster. This configuration is a set of properties that define the desired state of the cluster.
+
+- **Cluster app** refers to a Helm chart that defines the resources to be created in the workload cluster. The chart is maintained by Giant Swarm and is versioned.
+
+## Background
+
+So far, Giant Swarm users created workload clusters mainly using these two methods:
+
+- Via the `kubectl gs template cluster` CLI command
+- Via the web user interface (happa)
```

Comment:
More exactly it is `kubectl gs gitops add workload-cluster`.

Since it has been asked how this spec helps in a GitOps scenario, one open task is still to address that question.